### PR TITLE
bump package versions, add git installed versions

### PR DESCRIPTION
**Pull request by scottyhq**
* icepyx=0.6.2 (cc @JessicaS11)
* earthdata=0.2.0 (cc @mikala-nsidc , @betolink)
* nisardev, grimpfunc (cc @fastice)

### CODE REVIEW

File: conda/lock-environment.sh [hunk 1]
```diff
@@ -32,11 +32,15 @@ printf "Generate environment.yml for BinderHub \n"
...
+sed -i.bak -r 's/--hash=md5:None//' conda-linux-64.lock.yml
 sed -i.bak -r 's/--hash=md5:None//' conda-osx-64.lock.yml
 rm *.yml.bak
 
-cp conda-linux-64.lock.yml ../binder/environment.yml
+# Temporary fix for https://github.com/conda-incubator/conda-lock/issues/171
+echo "    - git+https://github.com/fastice/nisardev.git@4fa6429a93b6716e4e7a6522d98340b9966d7d9d" >> conda-linux-64.lock.yml
+echo "    - git+https://github.com/fastice/nisardev.git@4fa6429a93b6716e4e7a6522d98340b9966d7d9d" >> conda-osx-64.lock.yml
+echo "    - git+https://github.com/fastice/grimpfunc.git@512a8d35731c83771e74cbd5b0ade273fff680bd" >> conda-linux-64.lock.yml
+echo "    - git+https://github.com/fastice/grimpfunc.git@512a8d35731c83771e74cbd5b0ade273fff680bd" >> conda-osx-64.lock.yml
```
Comment (scottyhq) [inline conda/lock-environment.sh:43]: https://github.com/ICESAT-2HackWeek/website2022/issues/55

File: conda/lock-environment.sh [hunk 2]
```diff
@@ -32,11 +32,15 @@ printf "Generate environment.yml for BinderHub \n"
 conda-lock render -k env
 
 # Temporary fix for https://github.com/conda-incubator/conda-lock/issues/172
-sed -i.bak -r 's/--hash=md5:None//' conda-linux-64.lock.yml 
+sed -i.bak -r 's/--hash=md5:None//' conda-linux-64.lock.yml
 sed -i.bak -r 's/--hash=md5:None//' conda-osx-64.lock.yml
 rm *.yml.bak
 
-cp conda-linux-64.lock.yml ../binder/environment.yml
```
Comment (scottyhq) [inline conda/lock-environment.sh:39]: this is now just a softlink in binder directory (`ln -s ../conda/conda-linux-64.lock.yml environment.yml`)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ Ready to contribute? Here's a quick guide
 
 ## Contributing website changes
 
+For our Python environment management we recommend using `conda` and `mamba`. If you already do not already have those tools installed, you can use the [Mambaforge Installer for your platform](https://github.com/conda-forge/miniforge#mambaforge)
+
```
Comment (jomey) [inline CONTRIBUTING.md:15]: I suggest to link to the ReadMe with no specific heading selected. Gives the user a chance to read more before selecting a certain install package they prefer

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -28,10 +30,10 @@ Ready to contribute? Here's a quick guide
 
 1. Create and activate the "hackweek" conda environment.
 
-   __NOTE__: If you're running linux or Windows use `conda/conda-linux-64.lock`
-
+   __NOTE__: If you're running linux or Windows use `conda/conda-linux-64.lock.yml`
+   __NOTE__: `mamba env create` tends to be much faster than `conda env create`
     ```sh
-    conda create --name hackweek --file conda/conda-osx-64.lock
+    mamba env create --name hackweek --file conda/conda-osx-64.lock.yml
```
Comment (jomey) [inline CONTRIBUTING.md:36]: I am little concerned that a first time user will get confused mixing and matching the `conda` and `mamba` commands. Even if they basically do the same, from my perspective, a sentence explaining that would help make it more digestible.

File: conda/README.md [hunk 5]
```diff
@@ -9,5 +9,7 @@ lockfiles ensure everyone working on this project has exactly the same developme
 ```
 conda deactivate
 conda remove --name hackweek --all
-conda create --name hackweek --file conda-linux-64.lock
+# Can use mamba to speed this step up:
```
Comment (jomey) [inline conda/README.md:12]: See comments for the CONTRIBUTING regards the mixing and matching of commands. Explaining the difference ahead of time will spare having to comment this line.

File: conda/lock-environment.sh [hunk 6]
```diff
@@ -9,7 +9,7 @@ LOCK_ENV='CondaLock'
 conda env list | grep ${LOCK_ENV} > /dev/null
 
 if [[ $? -eq 1 ]]; then
-  conda create -q -y -n ${LOCK_ENV} -c conda-forge conda-lock=1 mamba=0.22
+  conda create -q -y -n ${LOCK_ENV} -c conda-forge conda-lock=1.0.3 mamba=0.22
```
Comment (jomey) [inline conda/lock-environment.sh:12]: Remind me again why mamab is a pinned version here?

**Review by jomey — APPROVED** [review]

File: conda/lock-environment.sh (hunk 6 above)
Comment (scottyhq) [reply conda/lock-environment.sh:12]: mamba isn't a required dependency of conda-lock (`conda search conda-lock --info`) I think keeping the locking environment as pinned as possible is wise.

File: CONTRIBUTING.md (hunk 3 above)
Comment (scottyhq) [reply CONTRIBUTING.md:15]: going to leave this b/c it's what we use in CI and guarantees mamba comes along for the ride with conda

**Comment by scottyhq** [comment]
Thanks @jomey all good points. In the last commit I switch to use just `mamba` instead of `conda` and added more narrative explanation at the top. I'll consider it a huge win if we get some contributions following these instructions :) !

**Pull request merged**